### PR TITLE
Cater for new OT classes API

### PR DESCRIPTION
**Pull request by alshakero**
JSON-Patch-OT classes API are about to change (if approved) and this caters for the upcoming changes in https://github.com/Palindrom/JSON-Patch-Queue/pull/7 and https://github.com/Palindrom/JSON-Patch-OT-agent/pull/4

Naturally, tests won't pass until the mentioned PRs are merged.

Fixes: https://github.com/Palindrom/Palindrom/issues/157

### CODE REVIEW

**Comment by warpech** [comment]
Looks like you've nailed the reason for the error, but I am not convinced about the robustness of the solution. 

Is it possible to solve it by changing the first argument of the `receive()` method from `obj` reference to a callback that returns `obj`? That would be easier to use correctly and more similar to the previous API.

File: test/specs/dom/ot.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
...,"WebsiteProvider_0":{"Html":"/WebsiteProvider/views/ContentWrapperPage.html","Content":{"People_0":{"Html":"/People/viewmodels/MasterPage.html","ShowMenu":true,"CurrentPage":{"People_0":{"Html":"/People/viewmodels/OrganizationsPage.html","Organizations":[],"AddUrl":"/people/organizations/add","RedirectUrl$":"","Confirm":{"Html":"/People/viewmodels/ConfirmDialogPage.html","Message":"","Ok$":"","Reject$":""},"EntriesPerPage$":0,"Pagination":{"Html":"/People/viewmodels/PaginationPage.html","ChangePage$":0,"NextPage$":0,"PreviousPage$":0,"LastPage$":0,"FirstPage$":0,"EntriesPerPage":5,"PageEntries":[{"Amount":5,"Text":"Show 5 items per page"},{"Amount":15,"Text":"Show 15 items per page"},{"Amount":30,"Text":"Show 30 items per page"}],"Pages":[],"TotalEntries":0,"TotalPages":0,"CurrentPage":1,"CurrentOffset":0,"DisableFirst":true,"DisableLast":false}}}}}}},"TopBar":{"WebsiteProvider_1":{},"WebsiteProvider_0":{},"SignIn_0":{"Uri":"","Html":"/SignIn/viewmodels/SignInPage.html","IsSignedIn":true,"Message":"","FullName":"admin admin","SignInClick$":0,"Submit":0,"SessionUri":"/__default/7F3049B0CEE1FA3530000000","UserImage":{"SignIn_0":{"Html":"/SignIn/viewmodels/UserImagePage.html"}}}}}}}`
```
Comment (tomalec) [inline test/specs/dom/ot.js]: I'd reduce this JSON for sake of maintenance and readability of the test.

**Comment by alshakero** [comment]
All PRs updated and ready for review.

**Comment by alshakero** [comment]
@warpech could please review `OT.js`? You understand it the best. If you want to review the whole PR, it's only `ot.js` and `src/palindrom.js`

File: src/palindrom.js [hunk 2]
```diff
@@ -472,20 +472,21 @@ var Palindrom = (function() {
    * Non-queuing object that conforms JSON-Patch-Queue API
    * @param {Function} apply function to apply received patch
    */
-  function NoQueue(apply) {
+  function NoQueue(obj, apply) {
```
Comment (tomalec) [inline src/palindrom.js]: Please update jsdoc as well.

**Pull request merged**